### PR TITLE
added check for clusteroperators after upgrade to make sure no DEGRAD…

### PR DESCRIPTION
**Pull request by pruan-rht**
…ED condition exists

### CODE REVIEW

File: lib/openshift/cluster_operator.rb [hunk 1]
```diff
@@ -4,12 +4,22 @@ class ClusterOperator < ClusterResource
...
+
+    # return Hash of conditions for easier access
+    def conditions(user: nil, cached: true, quiet: false)
+      conditions = raw_resource(user: user, cached: cached, quiet: quiet).dig('status', 'conditions')
+      cond_hash = {}
+      conditions.each do |con|
+        cond_hash[con['type']] = con
+      end
+      cond_hash
+    end
```
Comment (akostadinov) [inline lib/openshift/cluster_operator.rb]: This method can go to `resource.rb` IMO.

File: features/step_definitions/operators.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# operators related helper steps
+Given /^all clusteroperators reached version #{QUOTED} successfully$/ do |version|
+  ensure_admin_tagged
+  clusteroperators = BushSlicer::ClusterOperator.list(user: admin)
+  clusteroperators.each do | co |
+    raise "version does not match #{version}" unless co.version_exists?(version: version)
```
Comment (akostadinov) [inline features/step_definitions/operators.rb]: What if one upgrades only single cluster operator? Should it just check all operators are non-degraded and then separate step to test particular operator is at version `X`? Or how does it work?

Comment (pruan-rht) [reply features/step_definitions/operators.rb]: @akostadinov this method for cluster-wide upgrade check.   I've not tried a single clusteroperator upgarde...not sure if that's possible (need to check).   As the method name indicates it's checking `all clusteropeators`.

Comment (pruan-rht) [reply features/step_definitions/operators.rb]: @akostadinov found out that the only way to update an individual `clusteroperator` is to create a release image with only one operator that was updated... so for non-ART folks, it's not possible to do that.  It's probably a good to check all of them anyway.

File: lib/openshift/cluster_operator.rb (hunk 1 above)
Comment (pruan-rht) [reply lib/openshift/cluster_operator.rb]: @akostadinov should've checked it earlier...there's already one [there](https://github.com/openshift/verification-tests/blob/master/lib/openshift/resource.rb#L283), but it just return the raw array of conditions.

File: features/step_definitions/operators.rb (hunk 2 above)
Comment (akostadinov) [reply features/step_definitions/operators.rb]: You sure we never get operators with different versions? Hard to believe for me. Can anybody else comment on such probability?

File: lib/openshift/cluster_operator.rb (hunk 1 above)
Comment (akostadinov) [reply lib/openshift/cluster_operator.rb]: You can create a method to return to you based on type. It is a big anti-pattern to override a method changing its contract (in this case return data structure). See [Liskov principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

Comment (akostadinov) [reply lib/openshift/cluster_operator.rb]: e.g. `oc.conditions_by_type("my_type", user: admin)`

File: features/step_definitions/operators.rb (hunk 2 above)
Comment (pruan-rht) [reply features/step_definitions/operators.rb]: @akostadinov Trevor states that `anyone can create a new release image; you just need a registry somewhere to push it to`.  For our upgrade workflow, we just do a cluster-wide upgrade and not individual clusteroperators.

File: lib/openshift/cluster_operator.rb (hunk 1 above)
Comment (pruan-rht) [reply lib/openshift/cluster_operator.rb]: @akostadinov apparently there's also an existing method `[condition](https://github.com/openshift/verification-tests/blob/master/lib/openshift/resource.rb#L290)`

**Comment by pruan-rht** [comment]
@akostadinov PR updated.  PTAL.  Thanks!

File: lib/openshift/cluster_operator.rb (hunk 1 above)
Comment (akostadinov) [reply lib/openshift/cluster_operator.rb]: great find!

File: features/step_definitions/operators.rb (hunk 2 above)
Comment (akostadinov) [reply features/step_definitions/operators.rb]: I'm not sure I completely agree here but we can understand better what we need once it is needed so ack.

**Pull request merged**